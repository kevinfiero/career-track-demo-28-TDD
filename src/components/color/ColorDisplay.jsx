import React from 'react';
import PropTypes from 'prop-types';

const ColorDisplay = ({ bgColor, fgColor, text }) => (
  <div 
    data-testid="display"
    style={{ 
      color: fgColor, 
      backgroundColor: bgColor, 
      width: '50rm', 
      height: '50rem' }}>
    <p style={{ fontSize: '5rem' }}>{text}</p>
  </div>
);

ColorDisplay.propTypes = {
  bgColor: PropTypes.string.isRequired,
  fgColor: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default ColorDisplay;
