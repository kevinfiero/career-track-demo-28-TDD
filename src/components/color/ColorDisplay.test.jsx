import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ColorDisplay from './ColorDisplay';

describe('ColorDisplay component', () => {
  afterEach(() => cleanup());
  it('renders ColorDisplay', () => {
    const { asFragment } = render(<ColorDisplay  
      bgColor="#FF0000" 
      fgColor="#0000FF" 
      text="Hello!" 
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
