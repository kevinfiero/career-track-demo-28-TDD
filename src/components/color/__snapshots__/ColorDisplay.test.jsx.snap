// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorDisplay component renders ColorDisplay 1`] = `
<DocumentFragment>
  <div
    data-testid="display"
    style="color: rgb(0, 0, 255); background-color: rgb(255, 0, 0); height: 50rem;"
  >
    <p
      style="font-size: 5rem;"
    >
      Hello!
    </p>
  </div>
</DocumentFragment>
`;
